frontend/catalyst/compilation_pipelines.py

```diff
@@ -837,6 +851,55 @@ def qjit(
             elements of this list are named sequences of MLIR passes to be executed. A ``None``
             value (the default) results in the execution of the default pipeline. This option is
             considered to be used by advanced users for low-level debugging purposes.
+        abstracted_axes (Optional(Union[
+                Sequence[Sequence[str]],
+                Dict[int, str],
+                Sequence[Dict[int, str]],
+            ])): This is an experimental feature.
+            A way to specify dynamic tensor shapes.
+            This option affects the parameters and the compilation of the annotated function.
+            Parameters with abstracted_axes will be compiled to ranked tensors with dynamic shapes.
+
+            There are three ways to use abstracted_axes. Passing a sequence of tuples:
+
+            .. code-block:: python
+
+                abstracted_axes=((), ('n',), ('m', 'n'))
+
+            Each tuple in the sequence corresponds to one of the parameters in the annotated
+            function. Each tuple corresponds to a parameter. The first tuple corresponds to the
+            first parameter, the second tuple to the second parameter, and so on. Empty tuples can
+            be used and correspond to parameters with statically known shapes.
+            Non-empty tuples correspond to parameters with dynamically known shapes.
+            The intended meaning is that the first argument will have a statically known shape,
+            the second argument will have the 0th axis be the symbolic
+            variable "n" and the third argument will have the 0th axis be the symbolic variable
+            "n" and the 1st axist will be the symbolic variable "m".
```

Comment:
```suggestion
            Each tuple in the sequence corresponds to one of the arguments in the annotated
            function. Empty tuples can
            be used and correspond to parameters with statically known shapes.
            Non-empty tuples correspond to parameters with dynamically known shapes.
            
            In this example above,
            
            - the first argument will have a statically known shape,
            
            - the second argument has its zeroth axis have dynamic
              shape ``n``, and
            
            - the third argument will have its zeroth axis with dynamic shape
              ``m`` and first axis with dynamic shape ``n``.
```

A few minor changes:

- Removed some duplicate information to be a bit more concise
- Used itemization to help with skim reading/clarity
- @erick-xanadu I removed the use of the phrase 'symbolic variable' here, as it can be a bit confusing to scientific Python developers (who may make a connection to systems like SymPy)